definitions-trimmed-migrate/src/services/azure-storage-service/AzureStorageService.ts

```diff
@@ -23,12 +25,105 @@ export class AzureStorageService {
             this.options.queueConnectionString,
             { retryOptions: { maxTries: 4 } }
         ).getQueueClient(this.options.queueName);
+
+        this.definitionsToLowercaseQueueClient = QueueServiceClient.fromConnectionString(
+            this.options.queueConnectionString,
+            { retryOptions: { maxTries: 4 } }
+        ).getQueueClient(this.options.queueDefinitionsToLowercaseName);
+    }
+
+    // probably should have been split with MigrationService
+    public async iterateAndQueueBlobsByType(componentType: string) {
+        const startTime = new Date();
+        let totalIterated = 0;
+        let totalQueued = 0;
+        let nextContinuationToken = undefined;
+
+        try {
+            for await (const blobResponse of this.containerClient
+                .listBlobsFlat({ prefix: componentType })
+                .byPage({ maxPageSize: 1000, continuationToken: nextContinuationToken })) {
+                const { continuationToken, segment } = blobResponse;
+                const blobsNamesToProcess: string[] = [];
+
+                if (continuationToken) {
+                    nextContinuationToken = continuationToken;
+                }
+
+                if (segment.blobItems.length) {
+                    const cachedBlobValues = await Promise.all(
+                        segment.blobItems.map(
+                            throat(50, (blob) => {
+                                const { name } = blob;
+
+                                return this.redisService.get(name);
+                            })
+                        )
+                    );
+
+                    cachedBlobValues.forEach((cachedBlobValue, i) => {
+                        const { name, properties } = segment.blobItems[i];
+                        const { lastModified } = properties;
+
+                        if (lastModified < startTime && cachedBlobValue === null) {
+                            blobsNamesToProcess.push(name);
+                        }
+                    });
+
+                    if (blobsNamesToProcess.length) {
+                        await Promise.all(
+                            blobsNamesToProcess.map(
+                                throat(50, (blobsNameToProcess) => {
+                                    this.logService.log(
+                                        'AzureStorageService.iterateAndQueueBlobsByType',
+                                        `Queued ${blobsNameToProcess}`
+                                    );
+
+                                    totalQueued++;
+
+                                    return this.queueMessage(blobsNameToProcess);
+                                })
+                            )
+                        );
+
+                        await Promise.all(
+                            blobsNamesToProcess.map(
+                                throat(50, (blobsNameToProcess) => {
```

Comment:
should `blobsNameToProcess` be 'blobNameToProcess', i.e. singular, and to avoid conflict with the blobsNamesToProcess on line 94?